### ISSUE TITLE
cell-var-from-loop false positive on lambda

### ISSUE THREAD

**Issue by holmanb** [pylint-dev/pylint#9557]
### Bug description

The cell-var-from-loop warning shows up on conditional use of SystemExit's code property. This appears similar to, but not the same as https://github.com/pylint-dev/pylint/issues/5508, which appears specific to keyword arguments.

Given the following code, run pylint on it. Note the comment.

```python
import sys
import pytest


def test_invalid_W0640():
    """
    This test demonstrates a false positive for W0640.

    Message emitted:
    Cell variable %s defined in loop

    Description:
    A variable used in a closure is defined in a loop. This will result in all
    closures using the same value for the closed-over variable.
    """
    def run_closure(closure):
        closure()

    for i in range(5):
        with pytest.raises(SystemExit) as e:
            # pylint incorrectly thinks that the following closure will cause
            # all runs to have to same value
            run_closure(lambda *_: sys.exit(i))
        assert e.value.code == i
```

### Command used

```shell
pylint test_pylint.py
```

### Pylint output

```python
************* Module test
test.py:23: [W0640(cell-var-from-loop), test_invalid_W0640.<lambda>] Cell variable i defined in loop

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 8.89/10, +0.00)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.12.2 (main, Apr  2 2024, 18:40:52) [GCC 13.2.0]
```

### OS / Environment

Ubuntu 24.04, pylint installed via pip

### Additional dependencies

pytest if you want to demonstrate that the code works as intended

**Comment by scorphus** [comment]
Looks like a duplicate of #7100.
